### PR TITLE
*Test. added Frontend unit test

### PR DESCRIPTION
**Pull request by xKeCo**

### CODE REVIEW

File: onecgiar-pr-client/src/app/pages/ipsr/pages/innovation-package-detail/pages/ipsr-innovation-use-pathway/pages/step-n2/pages/complementary-innovation/components/table-innovation/table-innovation.component.spec.ts [hunk 1]
```diff
@@ -109,10 +118,11 @@ describe('TableInnovationComponent', () => {
     const alertFeSpy = jest.spyOn(mockCustomizedAlertsFeService, 'show');
     const deleteComplementary = jest.spyOn(component.api.resultsSE, 'DELETEcomplementaryinnovation').mockReturnValue(of({}));
     const emitSpy = jest.spyOn(component.saveedit, 'emit');
-    component.Ondelete(id);
-    expect(alertFeSpy).toHaveBeenCalled();
-    expect(deleteComplementary).toHaveBeenCalledWith(id);
-    expect(component.status).toBe(false);
-    expect(emitSpy).toHaveBeenCalledWith(true);
+    component.Ondelete(id, () => {
```
Comment (gcmartinezg) [inline onecgiar-pr-client/src/app/pages/ipsr/pages/innovation-package-detail/pages/ipsr-innovation-use-pathway/pages/step-n2/pages/complementary-innovation/components/table-innovation/table-innovation.component.spec.ts:121]: @yecksin please take into consideration the way the methods are being named. `Ondelete` should not start with a capital letter. I recommend researching onto the linter rule [naming-convention](https://typescript-eslint.io/rules/naming-convention/)

File: onecgiar-pr-client/src/app/pages/ipsr/pages/innovation-package-detail/pages/ipsr-innovation-use-pathway/pages/step-n4/components/step-n4-bilateral-investment-table/modal/step-n4-edit-bilateral/step-n4-edit-bilateral.component.ts [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
+import { ApiService } from '../../../../../../../../../../../../shared/services/api/api.service';
 
 @Component({
   selector: 'app-step-n4-edit-bilateral',
   templateUrl: './step-n4-edit-bilateral.component.html',
   styleUrls: ['./step-n4-edit-bilateral.component.scss']
 })
 export class StepN4EditBilateralComponent implements OnInit, DoCheck {
-  @Input() body: any;
+  @Input() body: any = {};
```
Comment (gcmartinezg) [inline onecgiar-pr-client/src/app/pages/ipsr/pages/innovation-package-detail/pages/ipsr-innovation-use-pathway/pages/step-n4/components/step-n4-bilateral-investment-table/modal/step-n4-edit-bilateral/step-n4-edit-bilateral.component.ts:13]: @yecksin with @davinzifc we have decided not to allow the use of `any` as a type in the back-end. Please consider doing the same in the front-end, by adding the linter rule [no-explicit-any](https://typescript-eslint.io/rules/no-explicit-any)

File: onecgiar-pr-client/src/app/shared/components/navigation-bar/navigation-bar.component.ts [hunk 3]
```diff
@@ -14,9 +14,24 @@ import { ApiService } from '../../services/api/api.service';
...
+  emailAccess = [
+    'h.f.tobon@cgiar.org',
+    'admin@prms.pr',
+    'j.cadavid@cgiar.org',
+    'j.delgado@cgiar.org',
+    'd.casanas@cgiar.org',
+    'S.Galvez@cgiar.org',
+    'y.zuniga@cgiar.org',
+    'yecksin@gmail.com'
+  ];
```
Comment (gcmartinezg) [inline onecgiar-pr-client/src/app/shared/components/navigation-bar/navigation-bar.component.ts:26]: @yecksin @xKeCo may I know the purpose of these hard-coded emails here? If they are to validate stuff related to admin access, I don't think it is secure to have them written on code

File: onecgiar-pr-client/src/app/shared/data/internationalizationData.test.ts
Comment (gcmartinezg) [inline onecgiar-pr-client/src/app/shared/data/internationalizationData.test.ts:1]: @xKeCo @yecksin watch out for the way the files are being named. I would expect this file to be called `internationalization-data`

File: onecgiar-pr-client/src/app/shared/services/customized-alerts-fe.service.ts
Comment (gcmartinezg) [inline onecgiar-pr-client/src/app/shared/services/customized-alerts-fe.service.ts:1]: @xKeCo please consider renaming this as `AlertOptions` instead

File: onecgiar-pr-client/src/app/shared/services/export-tables.service.ts [hunk 4]
```diff
@@ -35,9 +37,11 @@ export class ExportTablesService {
         const workbook = { Sheets: { data: worksheet, 'TOC indicators by result': tocsheet }, SheetNames: ['data', 'TOC indicators by result'] };
         const excelBuffer: any = xlsx.write(workbook, { bookType: 'xlsx', type: 'array' });
         this.saveAsExcelFile(excelBuffer, fileName);
+        callback();
       });
     } catch (error) {
       this.customAlertService.show({ id: 'loginAlert', title: 'Oops!', description: 'Error generating file', status: 'error' });
+      callback();
```
Comment (gcmartinezg) [inline onecgiar-pr-client/src/app/shared/services/export-tables.service.ts:44]: @xKeCo as `callback()` is optional, what if when it has to be executed, the function is null/undefined?

**Review by gcmartinezg — CHANGES_REQUESTED** [review]
Please check the comments I've left on some files

File: onecgiar-pr-client/src/app/pages/results/pages/result-detail/pages/rd-theory-of-change/rd-theory-of-change.component.ts [hunk 5]
```diff
@@ -85,6 +98,9 @@ export class RdTheoryOfChangeComponent implements OnInit {
       error: err => {
         this.getConsumed = true;
         console.error(err);
+      },
+      complete: () => {
+        callback && callback();
```
[the remark this answers is not in the document]
Comment (yecksin) [reply onecgiar-pr-client/src/app/pages/results/pages/result-detail/pages/rd-theory-of-change/rd-theory-of-change.component.ts:103]: Understood

File: onecgiar-pr-client/src/app/shared/components/navigation-bar/navigation-bar.component.ts (hunk 3 above)
Comment (yecksin) [reply onecgiar-pr-client/src/app/shared/components/navigation-bar/navigation-bar.component.ts:26]: You're right, it was already deleted.

File: onecgiar-pr-client/src/app/pages/type-one-report/type-one-report.service.spec.ts
[the remark this answers is not in the document]
Comment (yecksin) [reply onecgiar-pr-client/src/app/pages/type-one-report/type-one-report.service.spec.ts:1]: @xKeCo, can you please validate this?

File: onecgiar-pr-client/src/app/shared/data/internationalizationData.test.ts
Comment (yecksin) [reply onecgiar-pr-client/src/app/shared/data/internationalizationData.test.ts:1]: Understood

Comment (yecksin) [reply onecgiar-pr-client/src/app/shared/data/internationalizationData.test.ts:1]: ok, it was renamed

File: onecgiar-pr-client/src/app/shared/services/customized-alerts-fe.service.ts
Comment (xKeCo) [reply onecgiar-pr-client/src/app/shared/services/customized-alerts-fe.service.ts:1]: Done ✅

**Review by gcmartinezg — APPROVED** [review]

File: onecgiar-pr-client/src/app/pages/type-one-report/type-one-report.service.spec.ts
[the remark this answers is not in the document]
Comment (xKeCo) [reply onecgiar-pr-client/src/app/pages/type-one-report/type-one-report.service.spec.ts:1]: The test was removed due to synchronization issues with the automation server, which were impacting the CI/CD process. This issue will be reviewed.

**Comment by gcmartinezg** [comment]
@xKeCo please let me know when you are done with the adjustments

**Comment by xKeCo** [comment]
> @xKeCo please let me know when you are done with the adjustments

The changes have been implemented and the comments have been addressed.

**Review by gcmartinezg — APPROVED** [review]

**Pull request merged**